airflow-core/src/airflow/config_templates/airflow_local_settings.py

```diff
@@ -315,6 +317,14 @@ def _default_conn_name_from(mod_path, hook_name):
             },
         }
 
+        if ELASTICSEARCH_WRITE_TO_ES:
+            REMOTE_TASK_LOG = ElasticsearchRemoteLogIO(
```

Comment:
The RemoteLogIO behaviour should be used regardless of the write to es setting. This RemoteLogIO is the preferred way of handling things, so we should pass WRITE_TO_ES as an argument to this class that controls the behaviour.